File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        if len(s) == 1:
            return 1

        reversed = s[::-1]
        idx = 0
        length = 0
        while reversed[idx] == " ":
            idx += 1
        for i in range(idx, len(reversed)):
            if reversed[i] == " ":
                break
            length += 1

        return length

File: 0058-length-of-last-word/0058-length-of-last-word.c
int lengthOfLastWord(char* s) {
    int l=strlen(s);
    int length = 0;
    int idx = l-1;

    if (l==1) return 1;

    while (s[idx] == ' ') {
        idx--;
    }

    for (int i=idx; i>=0; i--) {
        if (s[i]!=' ') length++;
        else break;
    }

    return length;
}